spec/lib/ssoe/configuration_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# rubocop:disable RSpec/SpecFilePathFormat
+
+require 'rails_helper'
+require 'ssoe/configuration'
+
+RSpec.describe SSOe::Configuration do
+  subject(:config) { described_class.send(:new) }
+
+  let(:base_url) { 'https://api.example.com/soap' }
+  let(:cert_path) { '/tmp/client.pem' }
+  let(:key_path) { '/tmp/client.key' }
+
+  let(:cert_data) { 'PEM-DATA' }
+  let(:key_data) { 'KEY-DATA' }
+  let(:cert_obj) { instance_double(OpenSSL::X509::Certificate) }
+  let(:key_obj) { instance_double(OpenSSL::PKey::RSA) }
+
+  let(:faraday_connection) { instance_double(Faraday::Connection) }
+
+  before do
+    allow(IdentitySettings.ssoe_get_traits).to receive_messages(url: base_url, client_cert_path: cert_path,
+                                                                client_key_path: key_path)
+
+    allow(File).to receive(:exist?).with(cert_path).and_return(true)
```

Comment:
If we do the above things I mentioned and add a local crt/key that isn't actually associated with anything, then we should be able to do this spec with far less mocking